### ISSUE TITLE
[T-SQL] CREATE USER {FOR|FROM} LOGIN not parsed

### ISSUE THREAD

**Issue by flipflop97** [sqlfluff/sqlfluff#5424]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

The following code snippets are not parsed:

```tsql
CREATE USER user_name FOR LOGIN login_name
```

```tsql
CREATE USER user_name FROM LOGIN login_name
```

### Expected Behaviour

They should be parsed, see:

https://learn.microsoft.com/en-us/sql/t-sql/statements/create-user-transact-sql?view=sql-server-ver16#syntax

### Observed Behaviour

The line is not parsed correctly:

```
== [stdin] FAIL                                                                                                                                                                                                                                      
L:   1 | P:  23 |  PRS | Line 1, Position 23: Found unparsable section: 'FOR
                       | LOGIN login_name'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Enter a command in the following form:

```bash
echo "CREATE USER user_name FOR LOGIN login_name" | sqlfluff lint -
```

See its parsing error:

```
== [stdin] FAIL                                                                                                                                                                                                                                      
L:   1 | P:  23 |  PRS | Line 1, Position 23: Found unparsable section: 'FOR
                       | LOGIN login_name'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

T-SQL

### Version

- sqlfluff, version 2.3.5
- Python 3.11.6

### Configuration

Happens on the basic configuration:

```
[sqlfluff]
dialect = tsql
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
